src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -74,6 +75,20 @@ export class ProductCard extends PureComponent {
 
     imageRef = createRef();
 
+    isConfigurableProductOutOfStock(productVariants) {
+        const variantsInStock = productVariants.filter((productVariant) => productVariant.stock_status === IN_STOCK);
+
+        return variantsInStock.length === 0;
+    }
+
+    isBundleProductOutOfStock(item) {
+        const { options } = item;
+
+        const optionsInStock = options.filter((option) => option.product.stock_status === 'IN_STOCK');
+
+        return optionsInStock.length === 0;
+    }
```

Comment:
This is not a function for component, rather container. Calculate values there, then pass to the component.